plugins/advisors/osv/src/main/kotlin/Osv.kt

```diff
@@ -139,46 +139,20 @@ class Osv(override val descriptor: PluginDescriptor, config: OsvConfiguration) :
     }
 }
 
-private fun createRequest(pkg: Package): VulnerabilitiesForPackageRequest? {
-    val name = when {
-        pkg.id.namespace.isEmpty() -> pkg.id.name
-        pkg.id.type == "Composer" -> "${pkg.id.namespace}/${pkg.id.name}"
-        else -> "${pkg.id.namespace}:${pkg.id.name}"
-    }
-
-    val ecosystem = when (pkg.id.type) {
-        "Bower" -> null
-        "Composer" -> Ecosystem.PACKAGIST
-        "Conan" -> Ecosystem.CONAN_CENTER
-        "Crate" -> Ecosystem.CRATES_IO
-        "Gem" -> Ecosystem.RUBY_GEMS
-        "Go" -> Ecosystem.GO
-        "Hackage" -> Ecosystem.HACKAGE
-        "NPM" -> Ecosystem.NPM
-        "NuGet" -> Ecosystem.NUGET
-        "Maven" -> Ecosystem.MAVEN
-        "Pub" -> Ecosystem.PUB
-        "PyPI" -> Ecosystem.PYPI
-        "Swift" -> Ecosystem.SWIFT_URL
-        else -> null
-    }
-
-    if (name.isNotBlank() && pkg.id.version.isNotBlank() && !ecosystem.isNullOrBlank()) {
-        return VulnerabilitiesForPackageRequest(
-            // Do not specify the purl here as it is mutually exclusive with the ecosystem.
-            pkg = org.ossreviewtoolkit.clients.osv.Package(
-                name = name,
-                ecosystem = ecosystem
-            ),
-            version = pkg.id.version
-        )
-    }
-
+private fun createRequest(pkg: Package): VulnerabilitiesForPackageRequest {
     // TODO: Support querying vulnerabilities by Git commit hash as described at https://osv.dev/docs/#section/OSV-API.
     //       That would allow to generally support e.g. C / C++ projects that do not use a dedicated package manager
     //       like Conan.
 
-    return null
+    // Work-around for missing purl converters, see https://github.com/google/osv.dev/issues/2402.
+    return if (pkg.id.type == "Swift") {
+        VulnerabilitiesForPackageRequest(
```

Comment:
Shall we keep the check for `name.isNotBlank() && pkg.id.version.isNotBlank()` ?

Reply:
For simplicity, I'd prefer to drop them. Nothing should go wrong without these checks in place.

Reply:
Could this lead to an "invalid request" response?

Reply:
I've checked, and `getVulnerabilityIdsForPackages()` returns a success-result with empty lists, not a failure-result, in the case of blank names or versions.

Reply:
Great, thanks!

Reply:
hm, but still. It makes requests which do not make sense / won't return results.
I'd prefer keeping these checks. However, I'll leave it up to you to move this forward.